server/src/routes/auth/signUp/signUp.js

```diff
@@ -9,26 +11,32 @@ import User from '../../../models';
 const signUp = async (req, res) => {
   const { username, email, password } = req.body;
 
+  //validation
+  const { error } = signupValidation(req.body);
+  if (error) return res.status(StatusCodes.BAD_REQUEST).send(error.details[0].message);
+
   // checking if this email is already in db
-  const emailExist = await User.findOne({ email });
-  if (emailExist)
-    return res.status(400).send({ status: 'fail', message: `An account for email ${email} already exist` });
+  const isEmailExists = await User.findOne({ email });
+  if (isEmailExists)
+    return res
+      .status(StatusCodes.BAD_REQUEST)
+      .send({ status: 'fail', message: `An account for email ${email} already exist` });
```

Comment:
please remove this status property from all places :)

Reply:
I'll do that, but IMO will be better to create controller and move logic to another layer of abstraction where it will be easy to write unit test. Then we could leave as it is with with 'res.satatus ' and write integration test of this api endpoint. If we remove status property of response this will be misleading because even if we return response with status fail and error custom message the default status response will be 200 which is wrong. If we try to create custom error handling in interceptor (frontend part) and try to figure out what is send from backend using common pattern 'response.status === 400 then do this...' we will have to switch for filtering our response in another field.